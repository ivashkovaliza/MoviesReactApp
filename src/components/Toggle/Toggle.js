import React, {Component} from "react";
import PropTypes from 'prop-types';
import {connect} from "react-redux";
import { bindActionCreators } from 'redux';
import { setFilter, moviesFetchData } from '../../actions/actions';
import './Toggle.scss';

class Toggle extends Component {
  constructor(props) {
    super(props);
    this.state = {
      active: this.props.toggleValues[0],
    }
  }

  setActiveClass(activeElem) {
    this.setState({
      active: activeElem,
    });

    //this.props.setFilter(activeElem);
    //this.props.fetchData('https://reactjs-cdp.herokuapp.com/movies');
  }

  render() {
    const firstTab = this.props.toggleValues[0];
    const secondTab = this.props.toggleValues[1];

    return (
      <div className={'sort'}>
        <p className={'sort__title'}>{this.props.title}</p>
        <button className={`sort__btn sort__btn--left ${this.state.active === firstTab ? 'sort__btn--active' : ''}`}
                onClick={() => {
                  this.props.handleToggleClick(firstTab);
                  this.setActiveClass(firstTab);
                }}>
          {this.props.toggleValues[0]}
        </button>
        <button className={`sort__btn sort__btn--right ${this.state.active === secondTab ? 'sort__btn--active' : ''}`}
                onClick={() => {
                  this.props.handleToggleClick(secondTab);
                  this.setActiveClass(secondTab);
                }}>
          {this.props.toggleValues[1]}
        </button>
      </div>
    );
  }
}

const mapStateToProps = (state) => {
  return {
    searchBy: state.searchBy,
    sortBy: state.sortBy,
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    setFilter: (filter) => dispatch(setFilter(filter)),
    fetchData: (url) => dispatch(moviesFetchData(url))
  }
};

export default connect(mapStateToProps, mapDispatchToProps)(Toggle);

Toggle.propTypes = {
  title: PropTypes.string,
  toggleValues: PropTypes.array,
  onToggle: PropTypes.func,
  setFilter: PropTypes.func,
  fetchData: PropTypes.func,
  searchBy: PropTypes.string,
  handleToggleClick: PropTypes.func
};
