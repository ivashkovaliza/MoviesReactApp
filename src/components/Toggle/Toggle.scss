@import '../../styles/_colors';

.sort {
  display: flex;
  align-items: center;

  &__title {
    margin-right: 20px;
    text-transform: uppercase;
  }

  &__btn {
    padding: 10px 20px;
    border: none;
    background-color: $transparent-grey;
    color: $white;
    text-transform: uppercase;

    &:hover {
      background-color: $transparent-grey;
    }

    &--active {
      background-color: $salmon;

      &:hover {
        background-color: $light-salmon;
      }
    }

    &--left {
      border-radius: 5px 0 0 5px;
    }

    &--right {
      border-radius: 0 5px 5px 0;
    }
  }
}
