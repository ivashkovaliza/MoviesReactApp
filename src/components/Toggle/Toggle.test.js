import React from 'react';
import Toggle from './Toggle';

describe('Toggle', () => {
  const toggleValues = ['title', 'genres'];
  const handleToggleClick = () => { };

  test('renders correctly', () => {
    const component = shallow(
      <Toggle toggleValues={toggleValues} />
    );

    expect(component).toMatchSnapshot();
  });

  test('Toggling is correct', () => {
    const component = shallow(
      <Toggle toggleValues={toggleValues} handleToggleClick={handleToggleClick} />
    );

    expect(component.find('.sort__btn--left.sort__btn--active').length).toBe(1);
    expect(component.find('.sort__btn--right.sort__btn--active').length).toBe(0);
    component.find('.sort__btn--right').simulate('click');
    expect(component.find('.sort__btn--right.sort__btn--active').length).toBe(1);
    expect(component.find('.sort__btn--left.sort__btn--active').length).toBe(0);
  });
});
