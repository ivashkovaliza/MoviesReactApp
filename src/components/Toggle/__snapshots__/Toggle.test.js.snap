// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toggle renders correctly 1`] = `
<div
  className="sort"
>
  <p
    className="sort__title"
  />
  <button
    className="sort__btn sort__btn--left sort__btn--active"
    onClick={[Function]}
  >
    title
  </button>
  <button
    className="sort__btn sort__btn--right "
    onClick={[Function]}
  >
    genres
  </button>
</div>
`;
